app/src/main/java/pm/gnosis/heimdall/ui/safe/pairing/PairingAuthenticator.kt

```diff
@@ -137,7 +137,7 @@ class PairingAuthenticatorViewModel @Inject constructor(
 
             val info =
                 try {
-                    gnosisSafeRepository.loadAuthenticatorInfo(safeInfo.owners[1])
+                    gnosisSafeRepository.loadAuthenticatorInfo(safeInfo.owners[0])
```

Comment:
Is there not a way how we can do that without "guessing" at what index the extension address is?

Reply:
we have to extend our db schema and save address mapings. It seems that currently there is no way how to tell which address in the owner list is a 2fa device

Reply:
SafeInfoDb could include 2fa address.

Reply:
You could just check if we have an authenticator info for each owner in the list and take the first we find

Reply:
I would not add it to the SafeInfoDb, since we would need to update over time (as we discussed before)

Reply:
see https://github.com/gnosis/safe-android/blob/master/app/src/main/java/pm/gnosis/heimdall/ui/transactions/view/helpers/SubmitTransactionHelper.kt#L232